src/RestApi/StoreApi/Utilities/ProductQuery.php

```diff
@@ -288,43 +283,148 @@ public function add_query_clauses( $args, $wp_query ) {
+		if ( $wp_query->get( 'min_price' ) || $wp_query->get( 'max_price' ) ) {
+			$args = $this->add_price_filter_clauses( $args, $wp_query );
 		}
 
-		if ( $wp_query->get( 'stock_status' ) ) {
-			$args['join']   = $this->append_product_sorting_table_join( $args['join'] );
-			$args['where'] .= $wpdb->prepare( ' AND wc_product_meta_lookup.stock_status = %s ', $wp_query->get( 'stock_status' ) );
+		return $args;
+	}
+
+	/**
+	 * Add in conditional price filters.
+	 *
+	 * @param array     $args Query args.
+	 * @param \WC_Query $wp_query WC_Query object.
+	 * @return array
+	 */
+	protected function add_price_filter_clauses( $args, $wp_query ) {
+		global $wpdb;
+
+		$adjust_for_taxes = $this->adjust_price_filters_for_displayed_taxes();
+		$args['join']     = $this->append_product_sorting_table_join( $args['join'] );
+
+		if ( $wp_query->get( 'min_price' ) ) {
+			$min_price_filter = $this->prepare_price_filter( $wp_query->get( 'min_price' ) );
+
+			if ( $adjust_for_taxes ) {
+				$args['where'] .= $this->get_price_filter_query_for_displayed_taxes( $min_price_filter, 'min_price', '>=' );
+			} else {
+				$args['where'] .= $wpdb->prepare( ' AND wc_product_meta_lookup.min_price >= %f ', $min_price_filter );
+			}
+		}
+
+		if ( $wp_query->get( 'max_price' ) ) {
+			$max_price_filter = $this->prepare_price_filter( $wp_query->get( 'max_price' ) );
+
+			if ( $adjust_for_taxes ) {
+				$args['where'] .= $this->get_price_filter_query_for_displayed_taxes( $max_price_filter, 'max_price', '<=' );
+			} else {
+				$args['where'] .= $wpdb->prepare( ' AND wc_product_meta_lookup.max_price <= %f ', $max_price_filter );
+			}
 		}
 
 		return $args;
 	}
 
 	/**
-	 * Add meta query.
+	 * Get query for price filters when dealing with displayed taxes.
 	 *
-	 * @param array $args       Query args.
-	 * @param array $meta_query Meta query.
-	 * @return array
+	 * @param float  $price_filter Price filter to apply.
+	 * @param string $column Price being filtered (min or max).
+	 * @param string $operator Comparison operator for column.
+	 * @return string Constructed query.
+	 */
+	protected function get_price_filter_query_for_displayed_taxes( $price_filter, $column = 'min_price', $operator = '>=' ) {
+		global $wpdb;
+
+		// Select only used tax classes to avoid unwanted calculations.
+		$product_tax_classes = $wpdb->get_col( "SELECT DISTINCT tax_class FROM {$wpdb->wc_product_meta_lookup};" );
+		$or_queries          = [];
+
+		// We need to adjust the filter for each possible tax class and combine the queries into one.
+		foreach ( $product_tax_classes as $tax_class ) {
+			$adjusted_price_filter = $this->adjust_price_filter_for_tax_class( $price_filter, $tax_class );
+			$or_queries[]          = $wpdb->prepare(
+				'( wc_product_meta_lookup.tax_class = %s AND wc_product_meta_lookup.`' . esc_sql( $column ) . '` ' . esc_sql( $operator ) . ' %f )',
+				$tax_class,
+				$adjusted_price_filter
+			);
+		}
```

Comment:
Clever! I was wondering how we'd account for products with different tax classes (or non taxable). So essentially this just ensures that we include the correct min/max amount for products in each registered tax class for the query right?

Reply:
Exactly!